### ISSUE TITLE
Sorting with Magic Item rarity doesn't work well.

### ISSUE THREAD

**Issue by Nedrapter** [League-of-Foundry-Developers/compendium-browser#28]
I've noticed two problems with this filter.
1) Many magical weapons do not appear on the list when you filter with their rarity. For example this weapon from Loot Tavern 
![image](https://user-images.githubusercontent.com/74311190/137658712-f244311d-0708-489c-b9ed-0a578eb1cbc5.png)
Which is magical and Uncommon, doesn't show up when you filter with "Uncommon" . You could filter by typing in the name "Uncommon" since this one has it on its name, but plenty of others do not.
2) Many armors that have uncommon, rare, legendary rarities are not considered "magical" which I assume the filter requires them to be. Of course it could still be the same problem as in case 1 and that's why they don't show up. Example armor
![image](https://user-images.githubusercontent.com/74311190/137658887-b608b8b3-aa34-4479-bde8-5253d919b98e.png)

**Comment by spetzel2020** [comment]
Widespread problem, probably related to this code:
```
        this.addItemFilter("Magic Items", "Rarity", 'data.rarity', 'select', 
        {
            Common: "Common",
            Uncommon: "Uncommon",
            Rare: "Rare",
            "Very rare": "Very Rare",
            Legendary: "Legendary"
        });
```
because the actual data value is RARE, LEGENDARY etc (all caps). However simply changing these values to uppercase doesn't work.

**Comment by spetzel2020** [comment]
Fixed in 0.7.2 (turns out the rarity is stored as camelcase: veryRare, rare etc.)
